nest-forms-backend/src/nases-consumer/subservices/email-forms.subservice.ts

```diff
@@ -36,19 +43,38 @@ export default class EmailFormsSubservice {
     private throwerErrorGuard: ThrowerErrorGuard,
     private prismaService: PrismaService,
     private mailgunService: MailgunService,
+    private oloMailerService: OloMailerService,
     private configService: ConfigService,
     private convertService: ConvertService,
     private formValidatorRegistryService: FormValidatorRegistryService,
   ) {
     this.logger = new LineLoggerSubservice('EmailFormsSubservice')
   }
 
-  // eslint-disable-next-line sonarjs/cognitive-complexity
-  async sendEmailForm(
+  /**
+   * Resolves the address based on the environment.
+   * If the address is a string, it returns the string.
+   * If the address is an object, it returns the production value if CLUSTER_ENV is 'production',
+   * otherwise it returns the test/staging value.
+   */
+  private resolveAddress(address: { test: string; prod: string }): string {
+    const isProd =
+      this.configService.get<string>('CLUSTER_ENV') === 'production'
+    return isProd ? address.prod : address.test
+  }
+
+  private getMailer(formDefinition: FormDefinitionEmail): Mailer {
+    return formDefinition.email.mailer === 'olo'
```

Comment:
I don't prefer ternary usage for these kind of statements where there are only two options, this says if Mailer is OLO choose OLO else (semantically in whatever else scenario) use mailgun. However you want to return mailgun only if mailer in definition is Mailgun, not in every "other" scenario. If you add a new mailer this is hard to track. (Not in this case, but when there are 10s of ternaries like this in the code)

I think this (or switch statement) is a better approach:
```
const mailers = [
  'olo': this.oloMailerService,
  'mailgun': this.mailgunService,
]
const mailer = mailers[formDefinition.email.mailer];
if (!mailer) {
  throw `Unsupported mailer: ${mailer}`;
}
return mailer;
```